apps/src/code-studio/pd/foorm/FoormLibraryEditorManager.jsx

```diff
@@ -23,81 +25,125 @@ const styles = {
-    formCategories: PropTypes.array,
+    libraryNamesAndVersions: PropTypes.array,
+    libraryCategories: PropTypes.array,
 
     // populated by redux
-    formQuestions: PropTypes.object,
-    availableForms: PropTypes.array,
-    resetAvailableForms: PropTypes.func,
+    libraryId: PropTypes.number,
+    availableLibraries: PropTypes.array,
+    resetAvailableLibraries: PropTypes.func,
+    availableLibraryQuestionsForCurrentLibrary: PropTypes.array,
+    resetAvailableLibraryQuestions: PropTypes.func,
     setLastSaved: PropTypes.func,
     setSaveError: PropTypes.func,
-    setFormData: PropTypes.func,
+    setLibraryQuestionData: PropTypes.func,
     setHasError: PropTypes.func,
-    setLastSavedQuestions: PropTypes.func
+    setLastSavedQuestion: PropTypes.func,
+    setLibraryData: PropTypes.func
   };
 
   constructor(props) {
     super(props);
 
     this.state = {
-      formKey: 0,
-      formPreviewQuestions: null,
       showCodeMirror: false,
       hasLoadError: false
     };
 
-    this.props.resetAvailableForms(this.props.formNamesAndVersions);
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.formNamesAndVersions !== this.props.formNamesAndVersions) {
-      this.props.resetAvailableForms(this.props.formNamesAndVersions);
-    }
+    this.props.resetAvailableLibraries(this.props.libraryNamesAndVersions);
   }
 
+  // could probably dedupe this method
   getFormattedConfigurationDropdownOptions() {
-    return this.props.availableForms.map((formNameAndVersion, i) => {
-      const formName = formNameAndVersion['name'];
-      const formVersion = formNameAndVersion['version'];
-      const formId = formNameAndVersion['id'];
-      return (
-        <MenuItem
-          key={i}
-          eventKey={i}
-          onClick={() => this.loadConfiguration(formId)}
-        >
-          {`${formName}, version ${formVersion}`}
-        </MenuItem>
+    return this.props.availableLibraries.map((libraryNameAndVersion, i) => {
+      const libraryName = libraryNameAndVersion['name'];
+      const libraryVersion = libraryNameAndVersion['version'];
+
+      return this.getMenuItem(
+        () => this.loadLibraryQuestions(libraryNameAndVersion),
+        `${libraryName}, version ${libraryVersion}`,
+        i
       );
     });
   }
 
-  loadConfiguration(formId) {
+  getFormattedLibraryQuestionDropdownOptions() {
+    return this.props.availableLibraryQuestionsForCurrentLibrary.map(
+      (libraryQuestionAndType, i) => {
+        const libraryQuestionName = libraryQuestionAndType['name'];
+        const libraryQuestionType = libraryQuestionAndType['type'];
+        const libraryQuestionId = libraryQuestionAndType['id'];
+
+        return this.getMenuItem(
+          () => this.loadLibraryQuestionContent(libraryQuestionId),
+          `${libraryQuestionName} (${libraryQuestionType})`,
+          i
+        );
+      }
+    );
+  }
+
+  getMenuItem = (clickHandler, textToDisplay, key) => {
```

Comment:
On LP we decided to start naming functions that return components with `render` so in this case the function would be called `renderMenuItem`

Reply:
Makes sense to me -- would you do the same for getFormattedLibraryQuestionDropdownOptions (not actually a valid React component b/c it's not wrapped in a div?)

Reply:
I'm going to guess you wouldn't, if the naming convention is supposed to be like this function could be executed and returned in a render() and there'd be no problem".